### DISCUSSION TITLE
Cannot resolve symbol 'DefaultDispatcherModule'

### DISCUSSION THREAD

**Discussion by ucmmesa** [openTCS/opentcs#42] [Q&A]
In RunKernel.java,four modules related to strategies are prompted that unable to be resolved.What's the reason and how to tackle this problem?

**Comment by swltr** [comment]
You will need to provide more information for anyone to be able to help you there. A minimum of helpful information would be:

- Which version of openTCS do you use?
- What is the actual problem/output/error message you're seeing?

**Reply by ucmmesa** [reply]
I use openTCS v5.6 with Intellij IDEA 2021.2.3.

What I see is the red wave line in RunKernel.java said that the symbol like  DefaultDispatcherModule  cannot be resolved.However,these modules are really located in the path by the import line.

![image](https://user-images.githubusercontent.com/11491673/188403015-2892d79d-ba5b-454c-b4d0-00b4bce9eb35.png)

**Reply by swltr** [reply]
This seems to be an IntelliJ IDEA issue. Unfortunately, I cannot tell you how to make IntelliJ properly resolve these classes. For NetBeans, [a line in `gradle/guice-project.gradle`](https://github.com/openTCS/opentcs/blob/e14261fb3a69a22058ffa5b8aa54313edb12e41e/gradle/guice-project.gradle#L27) was added quite a while ago. (I'm actually not sure whether it is still required. Anyway, resolution in NetBeans works fine here.)
